### DISCUSSION TITLE
How to manully close an EventStream

### DISCUSSION THREAD

**Discussion by RAPrograms** [rwf2/Rocket#2810] [Questions] [answered]
I am attempeting to close an EventStream![] from the server when a condition is met. I have tryed to return & break to stop it. However, it just restarts. So I am wondering if it is even possible to stop an event stream from the server.

**Comment by the10thWiz** [comment, marked as answer]
My understanding is that returning from the `EventStream` macro will close the stream once there is no more content to return. Is the client retrying the connection? Checking the MDN page (https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#closing_event_streams), the JS `EventSource` will automatically retry the connection if it gets dropped. Using `EventStream!`, an `Event::retry` can be sent to indicate how long the client should wait before attempting to reconnect, although there doesn't appear to be a way to prevent reconnection from the server side.

The likely solution for you is going to be to send a close event of some kind, indicating to the client that you will not be sending any more events, and they should close the connection. This way the client will not attempt to reconnect.

**Reply by RAPrograms** [reply]
So follow up question: Can I reject the inusual connection. Also thank you for the answer

**Reply by the10thWiz** [reply]
I believe so, by either returning an `EventStream` or another response type. If you don't need any specific behavior:

```rust
#[get("/")]
fn sse_route() -> Result<EventStream, Status> {
    Err(Status::Forbidden)
}
```

The second response type can be whatever you need it to be.

**Reply by RAPrograms** [reply]
Thank you so much
